Dockerfile

```diff
@@ -28,18 +28,30 @@ ARG STRIPE_API_KEY
 ARG STRIPE_WEBHOOK_SIGNING_SECRET
 ARG PHX_SERVER
 ARG DATABASE_URL
+ARG PLATFORM=amd64
 
-ARG BUILDER_IMAGE="hexpm/elixir:${ELIXIR_VERSION}-erlang-${OTP_VERSION}-debian-${DEBIAN_VERSION}"
+ARG BUILDER_IMAGE="hexpm/elixir-${PLATFORM}:${ELIXIR_VERSION}-erlang-${OTP_VERSION}-debian-${DEBIAN_VERSION}"
 ARG RUNNER_IMAGE="debian:${DEBIAN_VERSION}"
 
-FROM --platform=linux/amd64 ${BUILDER_IMAGE} as builder
+FROM --platform=linux/${PLATFORM} ${BUILDER_IMAGE} as builder_amd64
+ONBUILD RUN wget -O - 'https://github.com/elm/compiler/releases/download/0.19.1/binary-for-linux-64-bit.gz' | gunzip -c >/usr/local/bin/elm
 
+FROM --platform=linux/${PLATFORM} ${BUILDER_IMAGE} as builder_arm64
```

Comment:
Not sure I understand the tricks with `builder_amd64` and `builder_arm64` :sweat_smile:

Reply:
It wasn't very intuitive to make the sure that the platform uses the correct elm binary so I did some research and found that the best way to do it was to use separate build steps, 1 for each platform, and the only use that specific image for the given platform. It took a while to stumble upon [this article](https://medium.com/@tonistiigi/advanced-multi-stage-build-patterns-6f741b852fae) that helped me create this.

For example, if we use "PLATFORM=amd64", then the `builder` image will use `builder_amd64` since `builder` is aliasing `builder_${PLATFORM}`. `builder_amd64` will run the single line below it, downloading the elm binary from the official github as opposed to pulling it from the custom image that I created.

So it creates 2 possible images that we would use as `builder`, then uses the one that matches the platform.

Reply:
@loicknuchel I am actually unsure of the performance this may have on amd64 builds since it may still try to run the COPY under builder_arm64. I'll run a check on this tomorrow, and if its significantly slower, I may just have it download the binary instead of pulling it from the image. 

I didn't go that way initially since I don't love the idea of pulling a raw binary from a non-official repo, but we could do a SHA check and make sure it matches the one generated by that dockerfile I created.